From a8eb3581a3f2d0749127fff7ede6ad67a3a96ff2 Mon Sep 17 00:00:00 2001
From: Julien Gilli <julien.gilli@joyent.com>
Date: Wed, 14 Jun 2017 11:23:53 -0700
Subject: [PATCH] joyent/node-krill#10 createPredicate throws an error with
 erroneous message if type of field is not string, number or boolean

---
 package.json         | 2 +-
 tests/tst.invalid.js | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/package.json b/package.json
index dbcb356..f14b873 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "krill",
 	"description": "simple boolean filter language with support for C, DTrace and LDAP output",
-	"version": "1.0.0",
+	"version": "1.0.1",
 	"license": "MIT",
 	"contributors": [
 	    "Robert Mustacchi <rm@joyent.com>",
diff --git a/tests/tst.invalid.js b/tests/tst.invalid.js
index 753e74e..6f67f20 100644
--- a/tests/tst.invalid.js
+++ b/tests/tst.invalid.js
@@ -36,7 +36,9 @@ var invalids = [
     [ { 'lt': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
     [ { 'le': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
     [ { 'ge': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
-    [ { 'gt': [ 's', '3' ] },	/cannot be applied to fields of type/, types ]
+    [ { 'gt': [ 's', '3' ] },	/cannot be applied to fields of type/, types ],
+    [ { 'eq': [ 'foo', null ] },
+        /predicate { eq: \[ 'foo', null \] }: field "foo" is not a string, number, or boolean/ ]
     /* END JSSTYLED */
 ];
 
-- 
2.21.0

